### PR TITLE
Simple app that converts objects array from a json file using the codec library (+ log4j, tests)

### PR DESCRIPTION
**Pull request by martinmilev**
The jar file runs with 
> $java -jar messagepreview.jar -type json message.json --printAverageStats

.

### CODE REVIEW

**Comment by mgenov** [comment]
It looks that it's not working. 

```
mgenov:xml_json(messagepreview?) $ git log
499d78e - (HEAD -> messagepreview, origin/messagepreview) Simple app that converts objects array from a json file, using the codec library (+ log4j, tests). (30 minutes ago) <Martin Milev>
528af95 - (origin/vasil_mitov, origin/master, origin/martin_milev, origin/denis_dimitrov, origin/borislav_gadjev, origin/alexander_vladimirov, origin/alexander_hristov, origin/HEAD, master) Initial Commit (2 weeks ago) <Miroslav Genov>
mgenov:xml_json(messagepreview?) $ 
```

```
mgenov:xml_json(messagepreview?) $ gradle assemble
:compileJava
/Users/mgenov/temp/xml_json/src/main/java/com/clouway/messagepreview/JsonHandler.java:3: error: package com.clouway.codec does not exist
import com.clouway.codec.InvalidJsonException;
                        ^
/Users/mgenov/temp/xml_json/src/main/java/com/clouway/messagepreview/JsonHandler.java:4: error: package com.clouway.codec does not exist
import com.clouway.codec.JSONCodec;
                        ^
/Users/mgenov/temp/xml_json/src/main/java/com/clouway/messagepreview/JsonHandler.java:5: error: package com.fasterxml.jackson.core.type does not exist
import com.fasterxml.jackson.core.type.TypeReference;
                                      ^
/Users/mgenov/temp/xml_json/src/main/java/com/clouway/messagepreview/JsonHandler.java:6: error: package org.apache.log4j does not exist
import org.apache.log4j.BasicConfigurator;
                       ^
/Users/mgenov/temp/xml_json/src/main/java/com/clouway/messagepreview/JsonHandler.java:7: error: package org.apache.log4j does not exist
import org.apache.log4j.Logger;
                       ^
/Users/mgenov/temp/xml_json/src/main/java/com/clouway/messagepreview/JsonHandler.java:20: error: cannot find symbol
  private static Logger log = Logger.getLogger(JsonHandler.class.getName());
                 ^
  symbol:   class Logger
  location: class JsonHandler
/Users/mgenov/temp/xml_json/src/main/java/com/clouway/messagepreview/User.java:3: error: package com.fasterxml.jackson.annotation does not exist
import com.fasterxml.jackson.annotation.JsonCreator;
                                       ^
/Users/mgenov/temp/xml_json/src/main/java/com/clouway/messagepreview/User.java:4: error: package com.fasterxml.jackson.annotation does not exist
import com.fasterxml.jackson.annotation.JsonProperty;
```

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+sourceCompatibility = 1.8
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testCompile group: 'junit', name: 'junit', version: '4.11'
+    compile files('lib/codec.jar','lib/log4j-1.2.17.jar','lib/commons-lang3-3.4.jar')
```
Comment (mgenov) [inline build.gradle]: Doesn't look like a reasonable way for adding of dependencies.

Comment (martinmilev) [inline build.gradle]: I had errors with the maven repository.
I will find working libraries and add them. 
And will include the codec.

`> Could not resolve org.apache.commons:commons-lang3:3.4.
      > Could not get resource 'https://repo1.maven.org/maven2/org/apache/commons/commons-lang3/3.4/commons-lang3-3.4.pom'.
         > Could not GET 'https://repo1.maven.org/maven2/org/apache/commons/commons-lang3/3.4/commons-lang3-3.4.pom'.`

Comment (martinmilev) [inline build.gradle]: Done.

**Comment by mgenov** [comment]
```java
mgenov:libs(messagepreview) $ java -jar msgprv-1.0-SNAPSHOT.jar 
no main manifest attribute, in msgprv-1.0-SNAPSHOT.jar
mgenov:libs(messagepreview) $ 

```

**Comment by mgenov** [comment]
I don't see dependency resolution problems here: 

build.gradle
```
group 'com.clouway.messagepreview'
version '1.0-SNAPSHOT'

apply plugin: 'java'

sourceCompatibility = 1.8

repositories {
    mavenCentral()
}

dependencies {
    testCompile group: 'junit', name: 'junit', version: '4.11'
    compile group: 'log4j', name: 'log4j', version: '1.2.17'
    compile group: 'org.apache.commons', name: 'commons-lang3', version: '3.4'
    compile group: 'commons-codec', name: 'commons-codec', version: '1.10'
    compile group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: '2.7.5'
}

task fatJar(type: Jar) {
    manifest {
        attributes 'Implementation-Title': 'Gradle Jar File Example',
                'Implementation-Version': version
        attributes 'Main-Class': 'com.clouway.messagepreview.Main'
    }
    baseName = project.name + '-all'
    from { configurations.compile.collect { it.isDirectory() ? it : zipTree(it) } }
    with jar
}
```

```java
mgenov:xml_json(messagepreview!?) $ gradle clean assemble
:clean
:compileJava
/Users/mgenov/temp/xml_json/src/main/java/com/clouway/messagepreview/JsonHandler.java:3: error: package com.clouway.codec does not exist
import com.clouway.codec.InvalidJsonException;
                        ^
/Users/mgenov/temp/xml_json/src/main/java/com/clouway/messagepreview/JsonHandler.java:4: error: package com.clouway.codec does not exist
import com.clouway.codec.JSONCodec;
                        ^
/Users/mgenov/temp/xml_json/src/main/java/com/clouway/messagepreview/JsonHandler.java:48: error: cannot find symbol
    JSONCodec codec = new JSONCodec();
    ^
  symbol:   class JSONCodec
  location: class JsonHandler
/Users/mgenov/temp/xml_json/src/main/java/com/clouway/messagepreview/JsonHandler.java:48: error: cannot find symbol
    JSONCodec codec = new JSONCodec();
                          ^
  symbol:   class JSONCodec
  location: class JsonHandler
/Users/mgenov/temp/xml_json/src/main/java/com/clouway/messagepreview/JsonHandler.java:53: error: cannot find symbol
    } catch (InvalidJsonException e1) {
             ^
  symbol:   class InvalidJsonException
  location: class JsonHandler
5 errors
:compileJava FAILED
```

I'm not able to see other module than this one so it looks like that it's a problem with the structure cause you have added packet jar file directly (codec.jar) which doesn't look reasonable in the context of gradle.

**Comment by mgenov** [comment]
Multi-module projects are well described in: https://docs.gradle.org/current/userguide/multi_project_builds.html

**Comment by martinmilev** [comment]
It is the `task fatJar`.
I run it directly from the file in IDEA. So I've missed that it doesn't work properly.

**Comment by martinmilev** [comment]
Fixed the jar task.
I will see how to do it with modules.

**Pull request closed without merging**